extensions/wikia/FilePage/FilePageController.class.php

```diff
@@ -73,6 +73,92 @@ public function fileUsage() {
 		wfProfileOut( __METHOD__ );
 	}
 
+	/**
+	 * Collects data about what articles the current file appears in, either
+	 * locally and globally.
+	 *
+	 * @requestParam string type (optional) Possible values are 'global' or 'local', defaulting to 'local' if not given.
+	 *                                      Determines whether this collects a local or global file usage data
+	 */
+	public function fileRedir() {
+		wfProfileIn( __METHOD__ );
+		global $wgMemc;
+
+		$page = $this->getContext();
+
+		//fallback to main page
+		$url = Title::newMainPage()->getFullURL();
+		//wiki needs read privileges
+		if ( !$page->getTitle()->userCan( 'read' ) ) {
+			$this->url = $url;
+			return;
+		}
+		$redirKey = wfMemcKey( 'redir', $page->getTitle()->getPrefixedText() );
+
+		$displayImg = $img = false;
+		Hooks::run( 'ImagePageFindFile', [ $page, &$img, &$displayImg ] );
+		if ( !$img ) { // not set by hook?
+			$img = wfFindFile( $page->getTitle() );
+			if ( !$img ) {
+				$img = wfLocalFile( $page->getTitle() );
+			}
+		}
+
+		if ( !$img || $img && !$img->fileExists ) {
+			$wgMemc->add( $redirKey, $url );
+			$this->url = $url;
+			return;
+		}
+
+		$urlMem = $wgMemc->get( $redirKey );
+		if ( $urlMem ) {
+			$this->url = $urlMem;
+			return;
+		}
+
+		$res = $this->fetchLinks( $img->getTitle()->getDBkey() );
```

Comment:
IIRC this can result `false` so `foreach` below may have some issues when trying to iterate over it